platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java

```diff
@@ -89,14 +107,13 @@ public SwirldStateManager(
 
         Objects.requireNonNull(platformContext);
         Objects.requireNonNull(addressBook);
-        Objects.requireNonNull(selfId);
+        this.selfId = Objects.requireNonNull(selfId);
         this.stats = new SwirldStateMetrics(platformContext.getMetrics());
-        Objects.requireNonNull(statusActionSubmitter);
+        this.statusActionSubmitter = Objects.requireNonNull(statusActionSubmitter);
         this.softwareVersion = Objects.requireNonNull(softwareVersion);
+        this.time = Time.getCurrent();
```

Comment:
We shouldn't use `Time.getCurrent()` in components - instead you can get the time object stored inside of `PlaformContext`

Reply:
Fixed